### ISSUE TITLE
Using notEqual / noneOf with a MultiEntry index

### ISSUE THREAD

**Issue by poltak** [dfahlander/Dexie.js#697]
tl;dr: **Just wondering if the `WhereClause`'s `noneOf` and `notEqual` methods support multi-entry indexes?** I haven't seen anything on the docs saying otherwise. 

I saw they just use `inAnyRange` to exclude the given values which, looking at how the multi-entry index get laid out in IndexedDB, seems like it would still work.

I tried it out before with my Dexie setup, although it doesn't seem to exclude the values properly from the results. Here's a screenshot to explain what I mean (and in-case I'm doing something obviously wrong). `storage` being a `Dexie` instance, `pages` being a table with at least a `terms` multi-entry index
<img width="891" alt="screen shot 2018-05-08 at 11 40 19" src="https://user-images.githubusercontent.com/1130716/39738705-cf5f626c-52b6-11e8-8831-7f9423bc022c.png">

`noneOf`/`notEqual` seem to work fine on standard string indexes for me on that table and others.

If you think it may be a bug, I can try to reproduce it in a simple HTML page.

Thanks for taking the time to read.

**Comment by dfahlander** [comment]
The way multiEntry works in IndexedDB implies that the logical result is not as you would suppose it to be.

For example if you have {term: ["foo"]}, it you would get a match for notEqual("dexie")
If you have {term: ["foo", "bar"]}, you would get two matches for notEqual("dexie") with same obj.
If you have {term: ["foo", "bar", "dexie"]}, you would still get two matches.

So basically, the notEqual/noneOf operators will not behave intuitively on multiEntry indexes.

Dexie does not really do any "smart logic" on multiEntry indexes as of current versions (1.x, 2.x and 3.x).

As noneOf and notEqual anyway does not really benefit that much of their index use, they would probably be equally fast using a manual filter instead.

**Comment by poltak** [comment]
Thanks for the reply @dfahlander. That's cleared it up for me then 👍 I have managed to implement the intended behaviour by adding a `.filter`/`.and` clause to my terms lookup, to go through each result and exclude any where excluded terms are present. Bit slower than standard lookups, but it works and can be conditionally applied to not impact regular (no-excluded terms) search.
